numba/tests/test_tuples.py

```diff
@@ -361,6 +361,18 @@ def test_le(self):
 
 class TestNamedTuple(TestCase, MemoryLeakMixin):
 
+    def test_exception(self):
```

Comment:
This might be named differently, like `test_exception_during_instantiation_with_non_iterable_types`, but maybe not as long. Maybe one of the  other core-devs has a better suggestion here.